src/radical/entk/task.py

```diff
@@ -29,1135 +114,229 @@ class Task(object):
-        self._post_exec  = list()
-        self._stage_on_error = False
-
-        self._lfs_per_process = 0
-        self._mem_per_process = 0
-        self._cpu_reqs        = {'processes'           : 1,
-                                 'process_type'        : None,
-                                 'threads_per_process' : 1,
-                                 'thread_type'         : None}
-        self._gpu_reqs        = {'processes'           : 0,
-                                 'process_type'        : None,
-                                 'threads_per_process' : 0,
-                                 'thread_type'         : None}
-
+        from_dict = from_dict or {}
+        if not isinstance(from_dict, dict):
+            raise ree.TypeError(expected_type=dict,
+                                actual_type=type(from_dict))
 
+        super().__init__()
 
-        # Data staging attributes
-        self._upload_input_data    = list()
-        self._copy_input_data      = list()
-        self._link_input_data      = list()
-        self._move_input_data      = list()
-        self._copy_output_data     = list()
-        self._link_output_data     = list()
-        self._move_output_data     = list()
-        self._download_output_data = list()
+        if not from_dict.get('uid'):
+            self['uid'] = ru.generate_id('task.%(counter)04d', ru.ID_CUSTOM)
+        self['state'] = res.INITIAL
 
-        # Name of file to write stdout and stderr of task
-        self._stdout = ""
-        self._stderr = ""
+        self.update(from_dict)
 
-        # Additional attributes that help in mapping tasks
-        # to cuds and cus to tasks
-        self._path      = None
-        self._exit_code = None
-        self._tags      = None
+    # --------------------------------------------------------------------------
+    #
+    def __eq__(self, other):
 
-        # Keep track of res attained
-        self._state_history = [res.INITIAL]
+        if not isinstance(other, Task):
+            return False
 
-        # Stage and pipeline this task belongs to
-        self._p_stage    = {'uid': None, 'name': None}
-        self._p_pipeline = {'uid': None, 'name': None}
+        # FIXME: review having attribute "name" here
+        return self['uid'] == other['uid'] and self['name'] == other['name']
 
-        # populate task attributes if so requesteed
-        if from_dict:
+    # --------------------------------------------------------------------------
+    #
+    def __ne__(self, other):
 
-            if not isinstance(from_dict, dict):
-                raise ree.TypeError(expected_type=dict,
-                                    actual_type=type(from_dict))
+        return not self == other
 
-            for k,v in from_dict.items():
-                self.__setattr__(k, v)
+    # --------------------------------------------------------------------------
+    #
+    def __hash__(self):
 
+        # FIXME: review having attribute "name" here
```

Comment:
agree - the UID itself should be sufficient.  Also, name might be unset (`None`) - what happens then, is that hashable?

Reply:
made changes to use `uid` only

> Also, name might be unset (`None`) - what happens then, is that hashable?

hashable, and just return same value